crates/sui-core/src/checkpoints/mod.rs

```diff
@@ -930,9 +1035,15 @@ impl CheckpointBuilder {
                 [(sequence_number, summary)],
             )?;
         }
+
+        self.state
```

Comment:
Please add detailed comments on why this is necessary.
If possible, tests would be nice too

Reply:
There are comments at the method impl - i can add some here as well.

This PR was motivated because the existing tests very quickly found crashes. Do you think its worth having dedicated tests as well? I'm not even sure how I'd write them.